triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala

```diff
@@ -0,0 +1,314 @@
+          .flatMap { client =>
+            val runner = new Runner(client, appId, config.party, config.dar)
+            val filter = runner.getTriggerFilter(config.triggerId)
+            runner
+              .queryACS(client, filter)
+              .map({ case (acs, offset) => QueriedACS(runner, filter, acs, offset) })
+          }
+      context.pipeToSelf(acsQuery) {
+        case Success(msg) => msg
+        case Failure(cause) => QueryACSFailed(cause)
+      }
+      queryingACS()
+    }
+}
+
+object Server {
+
+  sealed trait Message
+  private final case class StartFailed(cause: Throwable) extends Message
+  private final case class Started(binding: ServerBinding) extends Message
+  final case class GetServerBinding(replyTo: ActorRef[ServerBinding]) extends Message
+  final case object Stop extends Message
+
+  case class TriggerParams(identifier: Identifier, party: String)
+  implicit object IdentifierFormat extends JsonFormat[Identifier] {
+    def read(value: JsValue) = value match {
+      case JsString(s) => {
+        val components = s.split(":")
+        if (components.length == 3) {
+          val parsed = for {
+            pkgId <- PackageId.fromString(components(0))
+            mod <- DottedName.fromString(components(1))
+            entity <- DottedName.fromString(components(2))
+          } yield Identifier(pkgId, QualifiedName(mod, entity))
+          parsed match {
+            case Left(e) => deserializationError(e)
+            case Right(id) => id
+          }
+        } else {
+          deserializationError(s"Expected trigger identifier of the form pkgid:mod:name but got $s")
+        }
+      }
+      case _ => deserializationError("Expected trigger identifier of the form pkgid:mod:name")
+    }
+    def write(id: Identifier) = JsString("${id.packageId}:${id.qualifiedName}")
+  }
+  implicit val triggerParamsFormat = jsonFormat2(TriggerParams)
+
+  def apply(
+      host: String,
+      port: Int,
+      ledgerConfig: LedgerConfig,
+      dar: Dar[(PackageId, Package)],
+  ): Behavior[Message] = Behaviors.setup { ctx =>
+    // implicit val system: ActorSystem = ctx.system
+    // http doesn't know about akka typed so provide untyped system
+    implicit val untypedSystem: akka.actor.ActorSystem = ctx.system.toClassic
+    implicit val materializer: Materializer = Materializer(untypedSystem)
+    // implicit val ec: ExecutionContext = ctx.system.executionContext
+    implicit val esf: ExecutionSequencerFactory =
+      new AkkaExecutionSequencerPool("TriggerService")(untypedSystem)
+
+    var triggers: Map[UUID, ActorRef[TriggerActor.Message]] = Map.empty
+
+    val route = concat(
+      post {
+        // Start a new trigger given its identifier and the party it should be running as.
+        // Returns a UUID for the newly started trigger.
+        path("start") {
+          entity(as[TriggerParams]) { params =>
+            println(params)
```

Comment:
Is this a debug left-over? Maybe use logger instead?

Reply:
I jut removed it, it wasn’t really useful for anything.